user/syscalls.c

```diff
@@ -56,9 +100,22 @@ ssize_t write(int fd, const void *buf, size_t count) {
   return retval;
 }
 
-_off_t lseek(int __fildes, _off_t __offset, int __whence) {
-  errno = ENOSYS;
-  return (off_t)-1;
+_off_t lseek(int fd, _off_t offset, int whence) {
```

Comment:
So I've tried writing an asm macro that calls `__errno`. It needs stack to save `$ra` and the value returned from syscall. I did **not** test it, but it should be something along the lines of:
```
/* This implementation uses newlib-specific errno processing. */
#define DO_SYSCALL(num) \
    addiu $sp, $sp, -16 \
    sw $ra, 12($sp)     \
    li $v0, num         \
    syscall             \
    bgez $v0, 1f        \
    nop                 \
    sw $v0, 8($sp)      \
    jal __errno         \
    nop                 \
    lw $v1, 8($sp)      \
    negu $v1, $v1       \
    sw $v1, 0($v0)      \
    li $v0, -1          \
1:  lw $ra, 12($sp)     \
    addiu $sp, $sp, 16  \
    jr $ra              \
    nop
```
If you still consider it shorter and cleaner, then I'm fine with rewriting `syscals.c` to assembly.